src/cpp/include/openvino/genai/generation_handle.hpp

```diff
@@ -52,37 +53,53 @@ struct GenerationResult {
     GenerationStatus m_status = GenerationStatus::RUNNING;
 };
 
+// Represents already generated tokens of running generate() method.
+// E.g. typically generate() method consists of multiple step() which generate
+// token by token. This structure represents a vector of already generated tokens so far
+// for a given prompt.
```

Comment:
That's not exactly true. I would rephrase that to: "... a vector of tokens generated since the last read...". 
To be more specific it's always vector of one element if N == 1 and we are not using beam search. 
For N > 1 and/or beam search is used vector contains all generated tokens.